app/assets/javascripts/actions/story.js

```diff
@@ -26,27 +26,35 @@ export const editStory = (id, newAttributes) => ({
   newAttributes
 });
 
+export const setLoadingStory = (id) => ({
+  type: actionTypes.SET_LOADING_STORY,
+  id
+});
+
 export const updateStory = (storyId, projectId, options) =>
   (dispatch, getState, { Story }) => {
     const { stories } = getState();
     const story = stories.find((story) => story.id === storyId);
 
     if (story._editing._isDirty) {
+      dispatch(setLoadingStory(story.id))
       return Story.update(story._editing, projectId, options)
         .then((story) => {
-          dispatch(updateStorySuccess(story));
-        });
+          dispatch(updateStorySuccess(story))
+        })
+        .catch(() => dispatch(setLoadingStory(story.id)))
     }
     return dispatch(toggleStory(story.id));
   };
 
 export const deleteStory = (storyId, projectId) =>
-  (dispatch, getState, { Story }) =>
-    Story.deleteStory(storyId, projectId)
+  (dispatch, getState, { Story }) => {
+    dispatch(setLoadingStory(storyId))
+    return Story.deleteStory(storyId, projectId)
       .then(() => dispatch(deleteStorySuccess(storyId)))
-      .catch(
-        error => {
-          // TODO: dispatch an action to notify user on error
-          alert(error.message);
-        }
-      );
+      .catch( (error) => {
+        dispatch(setLoadingStory(storyId))
```

Comment:
Are we still gonna show the spinner if there is an error?

Reply:
I thought it would be good to add the loading so the user doesn't access the fields even if it's going to return an error, what do you think?

Reply:
@talyssonoc this line is to remove the `loading:true` when it fails. 
- line 52 sets it to true
- if this don't break deleteStorySuccess sets to false
- if it breakes 52 sets to false

Reply:
So `setLoadingStory` actually toggles the loading state of the story?